app/consumer/genesis.go

```diff
@@ -15,7 +28,130 @@ import (
 // object provided to it during init.
 type GenesisState map[string]json.RawMessage
 
+type (
+	// Transformation callback for a consumer genesis 'section' exported from a
+	// specific version of a provider
+	TransformationCallback func([]byte, client.Context) (json.RawMessage, error)
+
+	// TransformationMap defines the mapping from a version to a transformation callback
+	TransformationMap map[string]TransformationCallback
+)
+
+// Migration of consumer genesis content as it is exported from a provider version v2
```

Comment:
Prob need some clarification on the use of "v2" and "v3" here, it seems like the term is applied to both consumer and provider. Are we referring to ICS major versions?

Reply:
I was looking for a way how a user could find out what  module version is used on a provider chain for the provider implementation. I found the command `gaiad query upgrade module_versions` suitable for that. If you have another approach in mind pls let me know.
The above command returned the following on a real node:
```
...
{
  "name": "provider",
  "version": "2"
}
...
```